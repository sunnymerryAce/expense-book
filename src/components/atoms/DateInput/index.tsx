import React, { useState, useEffect } from "react";
import DateFnsUtils from "@date-io/date-fns"; // choose your lib
import { DatePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
import useStyles from "./style";
export interface DateInputProps {
  handleChange: (...props: any[]) => any;
  defaultTimestamp?: number;
}

const DateInput: React.FC<DateInputProps> = (props) => {
  const classes = useStyles();
  const [selectedDate, setDateChange] = useState(new Date());

  // 初回のみデフォルト日付をpropsから設定
  useEffect(() => {
    if (props.defaultTimestamp) {
      console.log(props.defaultTimestamp);
      setDateChange(new Date(props.defaultTimestamp));
    }
  }, []);

  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <DatePicker
        disableToolbar
        inputVariant="outlined"
        format="yyyy/MM/dd"
        value={selectedDate}
        onChange={(date): void => {
          setDateChange(date as Date);
          props.handleChange(date);
        }}
        variant="inline"
        label="日付"
        autoOk={true}
        className={`DateInput-datepicker ${classes.datePicker}`}
      />
    </MuiPickersUtilsProvider>
  );
};

export default DateInput;
