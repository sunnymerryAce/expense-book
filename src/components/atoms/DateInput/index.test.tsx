import React from "react";
import DateInput from "./";
import { shallow, mount } from "enzyme";

describe("<DateInput />", () => {
  let container: any = null;
  const onChangeFunction = jest.fn();

  beforeEach(() => {
    container = shallow(<DateInput handleChange={onChangeFunction} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  // 実行日付によってsnapshotが変わるので捨象
  // test("should match the snapshot", () => {
  //   expect(container.html()).toMatchSnapshot();
  // });

  test("should have proper defaultTimestamp", () => {
    container = mount(
      <DateInput
        handleChange={onChangeFunction}
        // 2020/06/25
        defaultTimestamp={1593056096000}
      />
    );
    expect(container.find(".MuiOutlinedInput-input").props()).toMatchObject({
      value: "2020/06/25",
    });
  });

  test("event handler should be triggered", () => {
    container
      .find(".DateInput-datepicker")
      .simulate("change", { target: { value: "" } });
    expect(onChangeFunction).toBeCalled();
  });
});
