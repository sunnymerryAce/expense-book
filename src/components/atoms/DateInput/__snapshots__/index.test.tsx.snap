// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInput /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root DateInput-datepicker makeStyles-datePicker-1\\" format=\\"yyyy/MM/dd\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled\\" data-shrink=\\"true\\">日付</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" readonly=\\"\\" value=\\"2020/05/13\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-62 PrivateNotchedOutline-legendNotched-63\\"><span>日付</span></legend></fieldset></div></div>"`;
